File: shadow/535.java
public class Codec {
    private Map<String, String> map = new HashMap<>();
    int i = 0;
    // Encodes a URL to a shortened URL.
    public String encode(String longUrl) {
        StringBuilder sb = new StringBuilder();
        sb.append("http://tinyurl.com/");
        sb.append(i);
        String shortUrl = sb.toString();
        map.put(shortUrl, longUrl);
        i++;
        return shortUrl;
    }

    // Decodes a shortened URL to its original URL.
    public String decode(String shortUrl) {
        return map.get(shortUrl);
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(url));

File: Mao/535.py
import string

class Codec:
    alphabet = string.ascii_letters + '01234569'

    def __init__(self):
        url2code = {}
        code2url = {}

    def encode(self, longUrl):
        """Encodes a URL to a shortened URL.
        
        :type longUrl: str
        :rtype: str
        """
        if longUrl not in url2code:
            code = ''.join([random.choice(Codec.alphabet) for i in xrange(6)])
            self.url2code[longUrl] = code
            self.code2url[code] = longUrl
        return 'http://tinyurl.com/'+self.url2code[longUrl]

    def decode(self, shortUrl):
        """Decodes a shortened URL to its original URL.
        
        :type shortUrl: str
        :rtype: str
        """
        return self.code2url['/'.split(shortUrl)[-1]]
        

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))
